lighthouse-cli/test/smokehouse/run-smoke.js

```diff
@@ -0,0 +1,191 @@
+  id: 'pwa2',
+  expectations: smokehouseDir + 'pwa2-expectations.js',
+  config: smokehouseDir + 'pwa-config.js',
+}, {
+  id: 'dbw',
+  expectations: 'dobetterweb/dbw-expectations.js',
+  config: smokehouseDir + 'dbw-config.js',
+}, {
+  id: 'redirects',
+  expectations: 'redirects/expectations.js',
+  config: smokehouseDir + 'redirects-config.js',
+}, {
+  id: 'seo',
+  expectations: 'seo/expectations.js',
+  config: smokehouseDir + 'seo-config.js',
+}, {
+  id: 'offline',
+  expectations: 'offline-local/offline-expectations.js',
+  config: smokehouseDir + 'offline-config.js',
+}, {
+  id: 'byte',
+  expectations: 'byte-efficiency/expectations.js',
+  config: smokehouseDir + 'byte-config.js',
+  batch: 'perf',
+}, {
+  id: 'perf',
+  expectations: 'perf/expectations.js',
+  config: 'lighthouse-core/config/perf-config.js',
+  batch: 'perf',
+}, {
+  id: 'ttci',
+  expectations: 'tricky-ttci/expectations.js',
+  config: 'lighthouse-core/config/default-config.js',
+}];
+
+/**
+ * Display smokehouse output from child process
+ * @param {{id: string, process: NodeJS.Process} || {id: string, error: Error & {stdout : NodeJS.WriteStream, stderr: NodeJS.WriteStream}}} result
+ */
+function displaySmokehouseOutput(result) {
+  console.log(`\n${purpleify(result.id)} smoketest results:`);
+  if (result.error) {
+    console.log(result.error.message);
+    process.stdout.write(result.error.stdout);
+    process.stderr.write(result.error.stderr);
+  } else {
+    process.stdout.write(result.process.stdout);
+    process.stderr.write(result.process.stderr);
+  }
+  console.timeEnd(`smoketest-${result.id}`);
+  console.log(`${purpleify(result.id)} smoketest complete. \n`);
+}
+
+/**
+ * Run smokehouse in child processes for selected smoketests
+ * Display output from each as soon as they finish, but resolve function when ALL are complete
+ * @param {Array<SmoketestDfn>} smokes
+ */
+async function runSmokehouse(smokes) {
+  const cmdPromises = [];
+  for (const {id, expectations, config} of smokes) {
+    console.log(`${purpleify(id)} smoketest starting…`);
+    console.time(`smoketest-${id}`);
+    const cmd = [
+      'node lighthouse-cli/test/smokehouse/smokehouse.js',
+      `--config-path=${config}`,
+      `--expectations-path=${expectations}`,
+    ].join(' ');
+
+    // The promise ensures we output immediately, even if the process errors
+    const p = execAsync(cmd, {timeout: 3 * 60 * 1000, encoding: 'utf8'}).then(cp => {
+      const ret = {id: id, process: cp};
+      displaySmokehouseOutput(ret);
+      return ret;
+    }).catch(err => {
+      const ret = {id: id, error: err};
+      displaySmokehouseOutput(ret);
+      return ret;
+    });
+
+    // If the machine is terribly slow, we'll run all smoketests in succession, not parallel
+    if (process.env.APPVEYOR) {
+      await p;
+    }
+    cmdPromises.push(p);
+  }
+
+  return Promise.all(cmdPromises);
+}
+
+/**
+ *
+ * @param {string[]} argv
+ * @return {Map<string, Array<SmoketestDfn>>}
```

Comment:
`string|undefined`? or give a default batch name/symbol so it doesn't fall back to undefined